### ISSUE TITLE
BigQuery: Question Stays running until timeout when query is error in Native Query

### ISSUE THREAD

**Issue by asyarif93** [metabase/metabase#14918]
**Describe the bug**
Question stays running until timeout when run broken query in native query with BigQuery Database (while in BQ itself already stopped with error). this happens since version 0.38.0 and 0.38.0.1.

**Logs**
No logs produced.

**To Reproduce**
1. Ask new Question
2. Select Native Query
3. Select BigQuery database as source
4. run query `this should be error.`
5. Question stays running for 5 minutes or more without returning error (in BQ dashboard run query already stopped)

**Expected behavior**
* In version 0.37.9 this quickly return error
* Also metabase logging error with messages like `ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: `

**Screenshots**
in version 0.38.0
![Screenshot from 2021-02-22 14-54-29](https://user-images.githubusercontent.com/30652154/108678683-02b7ca00-751e-11eb-9ab8-3896019bf1d0.png)
in version 0.37.9
![Screenshot from 2021-02-22 14-57-41](https://user-images.githubusercontent.com/30652154/108678918-562a1800-751e-11eb-85a6-baa31b5a7755.png)

**Information about your Metabase Installation:**
Using Metabase official docker on Ubuntu 20.04.
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.182 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.10+9",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.10",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.10+9",
    "os.name": "Linux",
    "os.version": "5.8.0-43-generic",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "bigquery"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-02-16",
      "tag": "v0.38.0",
      "branch": "release-x.38.x",
      "hash": "471aa51"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
* Unable to fix broken query if user have not access to google bigquery dashboard 
* No Logs Produced makes harder to debug

**Additional context**
Stays running for 5 minutes or more without returning error

**Comment by ConradIrwin** [comment]
We are also experiencing this, which is very frustrating for creating new queries (as you cannot tell if the query is working but slow, or will never work).


It is very easy to reproduce:

* set up a new BigQuery connection in metabase admin
* run a query with a syntax error, for example: `select ,`
*  _expected_: the query should fail within a few seconds
* _actual_: the "doing science" spinner keeps spinning until the load balancer times out and you get an error

It clearly is sending the query to BigQuery as we can see an error in the query history there; so the bug must exist when handling error responses from BigQuery.

As an aside, we recently upgraded from 0.37.? to 0.38 to get pivot tables; but are considering downgrading. Is it possible to downgrade to v0.37 after installing v0.38 without restoring from a backup of the metabase database?

**Comment by flamber** [comment]
@ConradIrwin Metabase does not officially support downgrading, but it should be possible between 0.38.0 and 0.37.9 - of course anything related to new functionality (pivot etc) would not work/be broken. Make a backup, downgrade, if it works then great, otherwise revert to 0.38.0.1 again.

**Comment by ConradIrwin** [comment]
Great! We'll miss the new features :(.

Please let me know if there's anything we can do to help debug or fix this.

From looking through the change log, it seems like not many commits touched the BigQuery module in the last few months, but several were large changes across the project:

```
git log --oneline  HEAD --not v0.37.9 modules/drivers/bigquery/src`
* b5e2c5185 Cam Saul Fix using a query that contains joins to a table with FK remaps as a source query (#14745) (11 days ago)
* 6312927aa Simon Belak (tag: v0.38.0-rc4) Nested queries revamp base (#14014) (4 weeks ago)
* 1f6d5fa35 Cam Saul Switch to flat namespace declarations (#14281) (7 weeks ago)
* 6fc2c60f5 dpsutton Microseconds support (:type/UNIXTimestampMicroseconds) (#1889) (#14024) (2 months ago)
* 6cd0ec7a8 Cam Saul Bern4rdelli bigquery cache per question (#13923) (3 months ago)
```

On a quick scan 1f6d5fa35 looks OK, as does 6fc2c60f5. 

* b5e2c5185 makes some changes to error handling in BigQuery and might be worth a more detailed look.
* 6cd0ec7a8 was the only change that was BigQuery specific, so also may be worth a more detailed look.
* 6312927aa calls out BQ in the commit message, but the only in-driver change was
```
-                                (sql.qp/build-honeysql-form driver)
+                                (sql.qp/mbql->honeysql driver)
```

**Comment by ConradIrwin** [comment]
(I tried to downgrade our staging metabase to `v0.38.0-preview` and `v0.38-rc1` to try and further narrow down the problem; but I couldn't get it to boot)

```
2021-02-23 18:12:37,079 ERROR metabase.core :: Metabase Initialization FAILED
liquibase.exception.ValidationFailedException: Validation Failed:
     2 change sets check sum
          migrations/000_migrations.yaml::275::dpsutton was: 8:447d9e294f59dd1058940defec7e0f40 but is now: 8:b976d49f4eb84664649119f5f63465dd
          migrations/000_migrations.yaml::276::robroland was: 8:59dd1fb0732c7a9b78bce896c0cff3c0 but is now: 8:dfe9b8333b600c739306caa85d95ad55

	at liquibase.changelog.DatabaseChangeLog.validate(DatabaseChangeLog.java:276) ~[metabase.jar:?]
	at liquibase.Liquibase.listUnrunChangeSets(Liquibase.java:1287) ~[metabase.jar:?]
	at liquibase.Liquibase.listUnrunChangeSets(Liquibase.java:1274) ~[metabase.jar:?]
	at liquibase.Liquibase.listUnrunChangeSets(Liquibase.java:1270) ~[metabase.jar:?]
```
)

**Comment by asyarif93** [comment]
sorry for late update, this is reproduce result in between 0.37.9 - 0.38.0
* v0.38.0-preview: returns error
* v0.38.0-rc1: returns error
* v0.38.0-rc2: returns error
* v0.38.0-rc3: returns error
* v0.38.0-rc4: returns error
* v0.38.0-rc5: returns error

**Comment by camsaul** [comment]
v0.38.0.1 is the latest version 😉

**Comment by camsaul** [comment]
Fixed by #15023
